gitgud/levels/rampup/_reversing/instructions.txt

```diff
@@ -1,19 +1,17 @@
 >>>
-The "git reset --soft <commit_ref>" is not used often. The branch pointer will move to the specified <commit_ref> (can use hash, HEAD^, HEAD~<num>, etc!). All the changes you've made since that commit will be left as staged ("Changes to be committed" in git status).
+Use "git reset --hard <commit_ref>" when you've made changes you do not want to keep. Using this mode will not preserve any changes you have made. A typical use of "git reset --hard" is when you've made changes to the wrong branch. After moving these changes to the correct branch, it is useful to use "git reset --hard" to return the incorrectly modified branch back to its original state.
 >>>
-The "git reset --mixed <commit_ref>" is the default mode of git reset. In other words "git reset <commit_ref>" implies the mixed mode. The functionality is similar to soft mode—the branch pointer will be moved to <commit_ref> and changes will be preserved. The only difference is that your changes will be unstaged (will need to use "git add").
+What if you want to discard some but not all changes you've made? Use "git reset --soft <commit_ref>" or "git reset <commit_ref>". These are the soft and mixed modes respectively. These commands will discard the working commit; however, the changes you've made will still be preserved as staged and unstaged changes respectively. In this way, you can selectively remove changes you dislike. Use a soft reset when most of the preserved changes will be kept (i.e. you will checkout/reset a smaller number of changes) and use a mixed reset when most of the preserved changes will be discarded (i.e. you only want to commit a couple of the preserved changes).
 >>>
-The third option is the most extreme (but may be the most useful). When you want to rollback the clock completely to a certain commit: use "git reset --hard <commit_ref>". This command will annihilate any changes you've made since the specified commit. Essentially, using this command will reset your repository to exactly what it was at the time of the specified commit. Be careful with this option!
+The second method is "git revert". Using "git revert" will create a new commit that will undo the changes that were made in a specified commit. For example, suppose I commit three files, 1, 2, and 3(<-HEAD). Using "git revert HEAD" will create a fourth commit where there are only files 1 and 2. In other words, the changes made at HEAD (commit 3) will be undone. Similarly, I could have used "git revert HEAD^" and this would have created a fourth commit with only files 1 and 3.
 >>>
-I mentioned "git reset" isn't the only revision command.
+Notice that in the previous example the commit with files 1, 2, and 3 still exists in the commit history! Git revert allows you to keep track of the commits prior to the revert. Therefore, if we decided we actually needed those changes, we can return to them. If you recall, this is unlike reset which abandons the working commit.
```

Comment:
Another important thing is that other people won't be able to see if you've deleted those commits, so git will just assume that they're the ones that made the changes and still make the unwanted changes. Reverting will undo the changes for everyone since they'll see the changes as a new commit, which can then be processed by git.

Reply:
Added